tasks/eva_3145/src/main/groovy/eva3145/eva3145_analysis.groovy

```diff
@@ -0,0 +1,193 @@
+import uk.ac.ebi.eva.groovy.commons.EVAObjectModelUtils
+import uk.ac.ebi.eva.groovy.commons.RetryableBatchingCursor
+
+import static org.springframework.data.mongodb.core.query.Query.query
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.*
+import static org.springframework.data.mongodb.core.query.Criteria.where
+
+// This script analyzes various metrics about variants with assembly match - see EVA-3145
+
+def cli = new CliBuilder()
+cli.prodPropertiesFile(args:1, "Production properties file for accessioning", required: true)
+cli.devPropertiesFile(args:1, "Development properties file for accessioning", required: true)
+cli.assembly(args:1, "Assembly to carry out the analysis for", required: true)
+def options = cli.parse(args)
+if (!options) {
+    cli.usage()
+    System.exit(1)
+}
+
+class EVA3145Analysis {
+    EVADatabaseEnvironment prodEnv
+    EVADatabaseEnvironment devEnv
+    String assembly
+    Logger scriptLogger
+    def cvesOfAsmMatchSSColl, cvesOfAsmMatchSSInOtherAsmColl, cvesMergedIntoAsmMatchColl, cvesMergedFromAsmMatchColl,
+            cvesSplitFromAsmMatchColl
+    def svesWithAsmMatchColl, svesSharingIDInOtherAsmColl, svesSharingIDWithoutAsmMatchColl, svesSplitFromAsmMatchSSColl,
+        opsForSvesMergedIntoAsmMatchSSColl
+    Map<String, Long> metricsMap = new HashMap<String, Long>()
+
+    EVA3145Analysis(options) {
+        this.prodEnv = createFromSpringContext(options.prodPropertiesFile, GenericApplication.class)
+        this.devEnv = createFromSpringContext(options.devPropertiesFile, GenericApplication.class)
+        this.assembly = options.assembly
+        this.scriptLogger = LoggerFactory.getLogger(GenericApplication.class)
+        (cvesOfAsmMatchSSColl, cvesOfAsmMatchSSInOtherAsmColl, cvesMergedIntoAsmMatchColl, cvesMergedFromAsmMatchColl,
+        cvesSplitFromAsmMatchColl,
+        svesWithAsmMatchColl, svesSharingIDInOtherAsmColl, svesSharingIDWithoutAsmMatchColl, svesSplitFromAsmMatchSSColl,
+        opsForSvesMergedIntoAsmMatchSSColl) =
+        ["cvesOfAsmMatchSS", "cvesOfAsmMatchSSInOtherAsm", "cvesMergedIntoAsmMatch", "cvesMergedFromAsmMatch",
+         "cvesSplitFromAsmMatch",
+         "svesWithAsmMatch", "svesSharingIDInOtherAsm", "svesSharingIDWithoutAsmMatch", "svesSplitFromAsmMatchSS",
+         "svesMergedIntoAsmMatchSSOps"]
+        def metrics = ["rs_of_asmmatch_ss", "rs_of_asmmatch_ss_in_other_asm",
+                       "rs_merged_into_asmmatch_rs", "rs_merged_from_asmmatch_rs", "rs_split_from_asmmatch_rs",
+                       "ss_with_asmmatch", "ss_sharing_id_in_other_asm", "ss_sharing_id_without_asmmatch",
+                       "ss_split_from_asmmatch", "ss_merged_into_asmmatch"]
+        metrics.each{metricsMap.put(it, 0)}
+    }
+
+    def runAsmMatchAnalysis() {
+        def dbsnpSveAsmMatchCursor = new RetryableBatchingCursor(where("seq").is(assembly)
+                .and("asmMatch").is(false), prodEnv.mongoTemplate, dbsnpSveClass)
```

Comment:
Currently, I have not excluded variants with mapWeight or allelesMatch flag because I wanted to assess the asmMatch variants in isolation. However, I have provided [stats](https://docs.google.com/spreadsheets/d/16t3T6XbZ3KSoZdfFyq7dsW2w8f-ZGVmAvMFNbEmd9T8/edit#rangeid=1124583023) on how many of the asmMatch variants have these two other flags.